syntax = "proto2";

package account;

import "gameTeamInfo.proto";
import "gameLoginData.proto";

//登录接口
message C2GS_Login{
 //channel  登陆的渠道号
	required int32 channel = 1;
 //uid
	required string uid = 2;
 //token
	required string token = 3;
 //md5
	optional string md5 = 4;
 //设备号
	optional string deviceId = 5;
 //
	optional string deviceType = 6;
 // 360用户id
	optional string qid = 7;
 // IMEI号
	optional string imei = 8;
}

message GS2C_Login{
	//md5
	optional string md5 = 1;
	//common 数据
	optional bytes commonData = 2;
	
	optional GameTeamInfo gameTeamInfoRespVo = 3;
	
	optional int64 nowTime = 4;
	//登陆时可变返回参数
	optional LoginData data = 5;
}

//登出接口
message C2GS_Logout{

}

message GS2C_Logout{

}

//获取经理人名字
message C2GS_GetUsername {
	
}
//返回经理人名字
message GS2C_GetUsername {
	optional string username = 1;
}

//注册俱乐部
message C2GS_RegisterClub {
	required int32 club = 1;
}
//返回注册俱乐部 -- code == 200
message GS2C_RegisterClub {
	optional int32 code = 1;
}

//创建角色 
message C2GS_CreateRole {
	//经理人名称
	required string name = 1;
}

//返回创建角色
message GS2C_CreateRole {
	optional GameTeamInfo gameTeamInfoRespVo = 1;
	//登陆时可变返回参数
	optional LoginData data = 2;
}

//异地登陆强制下线
message GS2C_KickOut {

}

//后台控制踢下线
message GS2C_KickOutByConsole {

}